src/allotropy/allotrope/schema_parser/generate_schemas.py

```diff
@@ -65,17 +66,17 @@ def files_equal(path1: str, path2: str) -> bool:
     return True
 
 
-def generate_schemas(root_dir: Path) -> int:
+def generate_schemas(root_dir: Path, *, dry_run: Optional[bool] = False) -> list[str]:
```

Comment:
nit: could you add a description of what `dry_run` is? also, the name was a little confusing to me, maybe like `validate` or `validation_run` or something like that

Reply:
Added a description for `dry_run`. `dry_run` is pretty commonly used for scripts/commands to mean run without saving changes, so I'm inclined to keep the name.